### PR TITLE
Remove Decision Tree and CatBoost Estimators from AutoML search

### LINKED ISSUE #4207: Remove CatBoost for all Problem Types

### LINKED ISSUE #4208: Remove DecisionTree for all Problem Types

### PR DESCRIPTION
**Pull request by christopherbunn**
Resolves #4207, Resolves #4208

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by jeremyliweishih — APPROVED** [review]
LGTM

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 1]
```diff
@@ -115,9 +113,9 @@ def test_all_estimators(
     is_using_conda,
 ):
     if is_using_conda:
-        n_estimators = 17
+        n_estimators = 13
```
[the remark this answers is not in the document]
Comment (jeremyliweishih) [reply evalml/tests/pipeline_tests/test_pipelines.py:116]: I don't believe so

File: docs/source/demos/lead_scoring.ipynb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
  "cells": [
   {
+   "attachments": {},
```
Comment (eccabay) [inline docs/source/demos/lead_scoring.ipynb:4]: Any idea why these got added here and removed from timeseries.ipynb?

**Review by eccabay — APPROVED** [review]

File: docs/source/demos/lead_scoring.ipynb (hunk 2 above)
Comment (christopherbunn) [reply docs/source/demos/lead_scoring.ipynb:4]: Ah, I think it's thrown in when you open the ipynb in VSCode. I thought I manually deleted some of these before I pushed. I'll take them out!

**Pull request merged**